cpp_tests/feature/one_hot_encoder_test.cc

```diff
@@ -14,79 +14,153 @@
+  EXPECT_TRUE(invalid_value_vector.IsError());
 
-  auto invalid_index_vector = model.transform(5, 0);
-  EXPECT_EQ(!invalid_index_vector.takeError(), false);
+  auto invalid_index_vector = model->transform(5, 0);
+  EXPECT_TRUE(invalid_index_vector.IsError());
 }
 
-TEST(OneHotEncoderTest, Load) {
-  std::string dir_name = clink::test::createTemporaryFolder();
+TEST_F(OneHotEncoderTest, Load) {
+  test::TemporaryFolder tmp_folder;
 
   nlohmann::json params;
   params["paramMap"]["dropLast"] = "false";
 
-  std::ofstream params_output(dir_name + "/metadata");
-  params_output << params;
-  params_output.close();
-
-  clink::OneHotEncoderModelDataProto model_data;
+  OneHotEncoderModelDataProto model_data;
   model_data.add_featuresizes(2);
   model_data.add_featuresizes(3);
 
-  mkdir((dir_name + "/data").c_str(), S_IRWXU);
-  std::ofstream model_data_output(dir_name + "/data/" +
-                                  clink::test::generateRandomString());
-  model_data.SerializeToOstream(&model_data_output);
-  model_data_output.close();
+  test::saveMetaDataModelData(tmp_folder.getAbsolutePath(), params, model_data);
+
+  auto model =
+      OneHotEncoderModel::load(tmp_folder.getAbsolutePath(), host_context);
+  EXPECT_FALSE((bool)model.takeError());
+
+  SparseVector expected_vector(2);
+  expected_vector.set(1, 1.0);
+  auto actual_vector = model.get()->transform(1, 0);
+  EXPECT_EQ(actual_vector.get(), expected_vector);
+}
 
-  auto model = clink::OneHotEncoderModel::load(dir_name);
-  EXPECT_EQ(!model.takeError(), true);
+TEST_F(OneHotEncoderTest, Mlir) {
+  test::TemporaryFolder tmp_folder;
 
-  auto vector = model->transform(1, 0);
-  EXPECT_EQ(!vector.takeError(), true);
-  EXPECT_EQ(vector->get(1).get(), 1.0);
+  nlohmann::json params;
+  params["paramMap"]["dropLast"] = "false";
+
+  OneHotEncoderModelDataProto model_data;
+  model_data.add_featuresizes(2);
+  model_data.add_featuresizes(3);
 
-  clink::test::deleteFolderRecursively(dir_name);
+  test::saveMetaDataModelData(tmp_folder.getAbsolutePath(), params, model_data);
+
+  // TODO: Separate the load process that is triggered only once and the
+  // repeatedly triggered transform process into different scripts.
+  auto mlir_script = R"mlir(
+    func @main(%path: !tfrt.string, %value: i32, %column_index: i32) -> !clink.vector {
+      %model = clink.onehotencoder_load %path
+      %vector = clink.onehotencoder_transform %model, %value, %column_index
+      tfrt.return %vector : !clink.vector
+    }
+  )mlir";
+
+  llvm::SmallVector<RCReference<AsyncValue>, 4> inputs;
+  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<std::string>(
+      tmp_folder.getAbsolutePath()));
+  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<int32_t>(1));
+  inputs.push_back(tfrt::MakeAvailableAsyncValueRef<int32_t>(0));
+
+  auto results =
+      test::runMlirScript(host_context, mlir_context, mlir_script, inputs);
+  EXPECT_EQ(results.size(), 1);
+  host_context->Await(results[0]);
```

Comment:
Would it be simpler to do `host_context->Await(results)` here? Same for other usages of Await(..).